packages/core/src/space/server/queries.ts

```diff
@@ -19,9 +19,21 @@ type FindSpaceBySlugInput = { slug: string };
 export const findSpaceBySlug = async (
   { slug }: FindSpaceBySlugInput,
   { db }: DbConfig,
-): Promise<Space | null> => {
+): Promise<(Space & { subspaces: Space[] }) | null> => {
   const results = await db.select().from(spaces).where(eq(spaces.slug, slug));
-  return results[0] || null;
+  const space = results[0];
+
+  if (!space) return null;
+
+  const subspaces = await db
+    .select()
+    .from(spaces)
+    .where(eq(spaces.parentId, space.web3SpaceId!));
```

Comment:
Please don't do this `non-null assertion`. By looking at your schema definition I'm getting confused here. You've defined:

```typescript
export const spaces = pgTable('spaces', {
  id: serial('id').primaryKey(),
  ...
  web3SpaceId: integer('web3_space_id'),
  ...
  parentId: serial('parent_id'),
  ...
});
```

It looks like you've intended to reference the `id` field. But instead you're query checks for `web3SpaceId`.